content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,436 @@
+---
+layout: post
+title: "Declarative Pipeline: Notifications and Shared Libraries"
+tags:
+- tutorial
+- pipeline
+- declarative
+- plugins
+- notifications
+- slack
+- hipchat
+- email-ext
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:https://cloudbees.com[CloudBees].
+
+**Declare Your Pipelines!**
+link:/blog/2017/02/03/declarative-pipeline-ga/[Declarative Pipeline 1.0 is here]!
+This is the third post in a series showing some of the cool features of
+link:/doc/book/pipeline/syntax/#declarative-pipeline[Declarative Pipeline].
+
+
+In the
+link:/blog/2017/02/10/declarative-html-publisher/[previous blog post],
+we converted Scripted Pipeline to a Declarative Pipeline with descriptive stages
+and `post` blocks.  In one of those `post&#123; }`  blocks, we included a placeholder for
+sending notifications.
+
+In this blog post, we'll pull in the notification method we created for
+link:/blog/2016/07/18/pipline-notifications/[Sending Notifications in Pipeline].
+Then we'll move that method into a shared library,
+letting us call it from other projects and keeping our `Jenkinsfile` concise and understandable.
+
+== Setup
+
+The setup for this post is a almost the same as the
+link:/blog/2017/02/10/declarative-html-publisher/[previous post].
+I've used a new branch
+link:https://github.com/bitwiseman/hermann[my same fork] of the
+link:https://github.com/reiseburo/hermann[hermann project]:
+link:https://github.com/bitwiseman/hermann/tree/blog/declarative/notifications[`blog/add-declarative/notifications`].
+I've set up a Multibranch Pipeline and pointed it at my repository
+the same as did it previous post.
+Also the same as before, I've set this Pipeline's Git configuration to
```

Comment:
To avoid the repetition of "also, also, additionally" I suggest joining this last sentence with the sentence prior, e.g.

> I have also set up a Multibranch Pipeline pointed at my repository, using the Pipeline's Git configuration to automatically "Clean after checkout", just as in my previous post.